src/zabapgit_object_prag.prog.abap

```diff
@@ -0,0 +1,176 @@
+*&---------------------------------------------------------------------*
+*&  Include  zabapgit_object_prag
+*&---------------------------------------------------------------------*
+
+CLASS lcl_object_prag DEFINITION INHERITING FROM lcl_objects_super FINAL.
+
+  PUBLIC SECTION.
+    INTERFACES lif_object.
+
+  PRIVATE SECTION.
+    TYPES: BEGIN OF ty_pragma,
+             pragma      TYPE c LENGTH 40,
+             extension   TYPE c LENGTH 1,
+             signature   TYPE c LENGTH 10,
+             description TYPE c LENGTH 255,
+           END OF ty_pragma.
+
+    METHODS:
+      _raise_pragma_not_exists
+        RAISING
+          lcx_exception,
+
+      _raise_pragma_exists
+        RAISING
+          lcx_exception,
+
+      _raise_pragma_enqueue
+        RAISING
+          lcx_exception.
+
+ENDCLASS.
+
+CLASS lcl_object_prag IMPLEMENTATION.
+
+  METHOD lif_object~has_changed_since.
+
+    rv_changed = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~changed_by.
+
+    rv_user = sy-uname.
+
+  ENDMETHOD.
+
+  METHOD lif_object~get_metadata.
+
+    rs_metadata = get_metadata( ).
+    rs_metadata-delete_tadir = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~exists.
+
+    TRY.
+        cl_abap_pragma=>get_ref( ms_item-obj_name ).
+
+      CATCH cx_abap_pragma_not_exists.
+        rv_bool = abap_false.
+        RETURN.
+    ENDTRY.
+
+    rv_bool = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~serialize.
+
+    DATA: lo_pragma TYPE REF TO cl_abap_pragma,
+          pragma    TYPE lcl_object_prag=>ty_pragma.
+
+    TRY.
+        lo_pragma = cl_abap_pragma=>get_ref( ms_item-obj_name ).
```

Comment:
do you know if this works on 702? as far as I remember pragmas was introduced sometime later